contracts/LiquidityMiningCampaign.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/utils/ReentrancyGuard.sol";
+import "./interfaces/IERC20Detailed.sol";
+import "./SafeERC20Detailed.sol";
+import "./RewardsPoolBase.sol";
+import "./LockScheme.sol";
+import "./StakeTransferer.sol";
+import "./StakeReceiver.sol";
+
+
+contract LiquidityMiningCampaign is RewardsPoolBase, StakeTransferer {
+	using SafeMath for uint256;
+    using SafeERC20Detailed for IERC20Detailed;
+
+	address[] lockSchemes;
+	address rewarsToken;
+	mapping(address => uint256) userAccruedRewads;
+
+	event StakedAndLocked(address indexed _userAddress, uint256 _tokenAmount, address _lockScheme);
+	event ExitedAndUnlocked(address indexed _userAddress, address _lockScheme);
+	event BonusTransferred(address indexed _userAddress, uint256 _bonusAmount);
+
+		constructor(
+        IERC20Detailed _stakingToken,
+        uint256 _startBlock,
+        uint256 _endBlock,
+        address[] memory _rewardsTokens,
+        uint256[] memory _rewardPerBlock,
+		uint256 _stakeLimit
+    ) public RewardsPoolBase(_stakingToken, _startBlock, _endBlock, _rewardsTokens, _rewardPerBlock,_stakeLimit) {
+
+		rewarsToken = _rewardsTokens[0];
+	}
+
+	function  stakeAndLock( uint256 _tokenAmount, address _lockScheme) external{
+		_stakeAndLock(msg.sender,_tokenAmount, _lockScheme);
+
+	}
+
+	function _stakeAndLock(address _userAddress ,uint256 _tokenAmount, address _lockScheme) internal {
+		require(_tokenAmount > 0, "stakeAndLock::Cannot stake 0");
+		require(!LockScheme(_lockScheme).hasUserRampUpEnded(_userAddress), "stakeAndLock::The ramp up period has finished");
+
+		UserInfo storage user = userInfo[_userAddress];
+
+		uint256 userRewards = 0;
+
+		if(user.tokensOwed.length > 0) {
+			userRewards = user.tokensOwed[0];
+		}
+	
+		userAccruedRewads[_userAddress]	= userAccruedRewads[_userAddress].add(userRewards);
```

Comment:
user accrued rewards should always be equal to userRewards; `userAccruedRewads[_userAddress] = userRewards`

Reply:
This update should always happen after the lock scheme rewards are accrued